addons/web/static/src/views/fields/badge_selection/badge_selection_field.js

```diff
@@ -63,6 +64,32 @@ export class BadgeSelectionField extends Component {
         return JSON.stringify(value);
     }
 
+    get spanClass() {
+        if (this.props.colorField && this.props.record.data[this.props.name]) {
+            return `badge rounded-pill o_badge_color_${
+                this.props.record.data[this.props.colorField]
+            }`;
+        }
+        return "";
+    }
+
+    badgeSelectionClass(value) {
+        const badgeClass = [];
+        if (["sm", "lg"].includes(this.props.size)) {
+            badgeClass.push(`btn-{this.props.size}`);
+        }
+        if (this.props.colorField) {
+            badgeClass.push("badge rounded-pill");
+            if (this.value === value) {
+                badgeClass.push("o_badge_border border");
```

Comment:
you could style it with the active class, no need for o_badge_border (nor border I think)